src/lib/y2network/hwinfo.rb

```diff
@@ -25,11 +25,68 @@ module Y2Network
   # FIXME: decide whether it should read hwinfo (on demand or at once) for a network
   # device and store only necessary info or just parse provided hash
   class Hwinfo
+    # TODO: this method should be private
     attr_reader :hwinfo
 
-    def initialize(name:)
+    class << self
+      # Creates a new instance containing hardware information for a given interface
+      #
+      # It retrieves the information from two sources:
+      #
+      # * hardware (through {Yast::LanItems} for the time being),
+      # * from existing udev rules.
+      #
+      # @todo Probably, this logic should be moved to a separate class.
+      #
+      # @param name [String] Interface's name
+      # @return [Hwinfo]
+      def for(name)
+        hwinfo_from_hardware(name) || hwinfo_from_udev(name) || Hwinfo.new
+      end
+
+    private
+
+      # Returns hardware information for the given device
+      #
+      # It relies on the {Yast::LanItems} module.
+      #
+      # @param name [String] Interface's name
+      # @return [Hwinfo,nil] Hardware info or nil if not found
+      def hwinfo_from_hardware(name)
+        hw = Yast::LanItems.Hardware.find { |h| h["dev_name"] == name }
+        return nil if hw.nil?
```

Comment:
this collides with https://github.com/yast/yast-network/pull/927/files#diff-d0c6d9ea3e0b7a970cde5a5b107ea158R124 . Ideally we should have class for that reading of hardware.

Reply:
Yes, I agree (something like `HardwareReader`), but it was of the scope of this PR. On the other hand, I will fix the collision.